### PR TITLE
SOLAS-5222: As a USER I am able to move to the next question

### PR DESCRIPTION
**Pull request by robinedman**
Based off of https://github.com/britishcouncil/react-quiz/pull/2. Until that one is merged you can see the changes [here](https://github.com/britishcouncil/react-quiz/compare/solas-5208-select-answer...solas-5222-next-question).

Makes it possible to move to the next question in the quiz by pressing `Next`. The last question has a `Finish` button.

Did some refactoring while completing this story. There's a "smart" component at the top (`QuizContainer`) which stores the quiz state. The components below are "dumb" functional components that receive `props` and return JSX markup. [More info](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0#.u2dio3y4p).
### Tips for reviewing

The tests are quite extensive this time. Reading through them can be a good way to learn how the code works. It's a good source for test cases that you can try manually as well.

[Internal JIRA story](https://bcdigital.atlassian.net/browse/SOLAS-5222)

### CODE REVIEW

File: demo/quiz.css [hunk 1]
```diff
@@ -0,0 +1,14 @@
+ul {
+  list-style: none;
+}
+
+.Quiz-answer {
```
Comment (2creative) [inline demo/quiz.css:5]: minor, we don't normally use upper case class names?

Comment (robinedman) [inline demo/quiz.css:5]: OK! It's following https://github.com/suitcss/suit/blob/master/doc/naming-conventions.md. I tend to use that in personal projects but I'm not really fussed about it.

I think we might end up removing all classes in the styling story and let the user of this component supply their own CSS classes. What do you say about leaving it to that story?

Comment (2creative) [inline demo/quiz.css:5]: Yea sure, sounds good..

**Comment by andreadsw** [comment]
Reviewing ...

**Comment by LucjaWszola** [comment]
Works for me. 👍 

quiz

Test passed.
test

**Comment by andreadsw** [comment]
@robinedman It looks good for me. 👍   I checked on deferents browsers such as Chrome, Firefox, Safari and IE9 above. It looks good. Also I checked on couple of devices as ipads and smartphones too. 

Thanks for explaining to me about {... } copying the contents of the object. I'm getting slowly my head around about these conventions.

**Comment by robinedman** [comment]
@LucjaWszola @andreadsw Thanks for the reviews! :)

**Pull request merged**